### DISCUSSION TITLE
deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#79] [Announcements]
# deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/

In this post I discuss various approaches to running database migrations when deploying to Kubernetes

https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-7-running-database-migrations/

**Comment by Mattes83** [comment]
Hi,
Great blog post, thank you for that. 
Regarding the Helm hook: There is a general timeout when using Helm install which defaults to 5 minutes, as described here: https://helm.sh/docs/intro/using_helm/#helpful-options-for-installupgraderollback
Could it be that this was the timeout you ran into?

If yes the helm hook might be the easier solution.

**Comment by ebuildy** [comment]
Hi,

My architecture:
- jobs via argo-workflows
- deployments wait for job status
- sync via a PVC or ConfigMap

1) Pod initContainer wait for condition (loop until PVC /syncho.txt is here)
2) Jobs write PVC synchro.txt

This is a better approach than using Hook.

You can even Deploy a "waiting page" behind the same service.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 02 2020, 06:50, in reply to [Dejan Lukić](https://disqus.com/by/disqus_8nhDyZqnbB/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/79#discussioncomment-6059588)</em>

---

<p>Check out post 9 in this series 😉</p>

[a comment by an automated account is not shown]

**Comment by ThiesR** [comment]
Hi Andrew, this is a really great post about how to do DB migrations in k8s. Thank you very much!

Cheers,
Thies

**Comment by davidkarlsen** [comment]
You can checkout this operator to do flyway migrations with a declarative API: https://github.com/davidkarlsen/flyway-operator
